### PR TITLE
Update cloud pricing page styles

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

This PR updates the styles of the pricing page in cloud, mostly to match jetpack.com typefaces.

Fixes 1169247016322522-as-1191918158300076

### Testing instructions

- Run Jetpack cloud and visit `/pricing`.
- For developers: check the code and that the page doesn't look broken.
- For designers: check that the page matches what's expected. Parts of the mockups seemed a bit outdated, so I used my best judgement.
- 

#### Mobile
screencapture-jetpack-cloud-localhost-3000-pricing-2020-09-04-16_33_46

#### Desktop
screencapture-jetpack-cloud-localhost-3000-pricing-2020-09-04-16_32_37

### CODE REVIEW

**Review by elliottprogrammer — APPROVED** [review]
Code looks good, page looks good!  LGTM! 👍

File: client/server/pages/index.js [hunk 1]
```diff
@@ -689,6 +690,12 @@ export default function pages() {
 				req.context.sectionGroup = section.group;
 			}
 
+			if ( Array.isArray( section.links ) ) {
+				section.links.forEach( ( link ) =>
+					req.context.store.dispatch( setDocumentHeadLink( link ) )
+				);
+			}
```
Comment (rcanepa) [inline client/server/pages/index.js:697]: This looks like a good idea. I wonder if there is another standard way of doing this.

**Comment by rcanepa** [comment]
Looks very nice and code is 👍 .

There are two things we still need to fix (@eeeeevon13 let me know if I'm right):

1. The primary color is Jetpack's green rather than WordPress's blue.
2. On a very small viewport, the Menu button should take the whole width of the screen (my bad for not implementing this when I created the navigation bar).

I think point 1. can be fixed by making sure color overrides from `wp-calypso/packages/calypso-color-schemes/src/shared/color-schemes/_jetpack-cloud.scss` are only applied when we are not inside the `.is-section-jetpack-cloud-pricing` class. So, it would be a matter of updating the first line from
```
.theme-jetpack-cloud,
```
to
```
.theme-jetpack-cloud:not( .is-section-jetpack-cloud-pricing ),
```

File: client/server/pages/index.js (hunk 1 above)
Comment (monsieur-z) [reply client/server/pages/index.js:697]: I'm not sure either. I'm not a big fan of having the font definition in `section.js`, as it feels a bit too specific for the purpose of this file. But I didn't want to add a condition that checks for the pricing page in a React component.

**Comment by monsieur-z** [comment]
> There are two things we still need to fix (@eeeeevon13 let me know if I'm right):

Sure thing. I'll do the update once @eeeeevon13 has reviewed the whole page.

**Review by eeeeevon13 — COMMENTED** [review]
First let me say, this is looking so good. Ya'll have done a really great job! Couple of things:

1. The action bar w/yearly monthly toggle & filter should be pinned when you scroll, that doesn't appear to be working https://drive.google.com/file/d/1nzhcu8TMS0fah_TF7qbRdkg7z4782p4D/view
2. Let's add a little more vertical space above and below the faq's. Increase to 40px to give the free plan some room to breathe.
3. The search card needs to be updated w/the new design/price/radio button
bordermobile
spacing

**Comment by monsieur-z** [comment]
As discussed in another channel, I'll merge this PR and address feedback in a new one, for testing purpose.

**Comment by monsieur-z** [comment]
Follow up PR: https://github.com/Automattic/wp-calypso/pull/45510

**Pull request merged**